### PR TITLE
Add architecture reference docs for authentication

### PR DESCRIPTION
**Pull request by simu**

### CODE REVIEW

**Comment by simu** [comment]
I'm not sure if the structure, location and naming makes sense. Naming is currently probably still a mix between "APPUiO Public 2.0 zone" and "OpenShift cluster".

File: docs/modules/ROOT/pages/references/architecture/openshift-idp-data-model.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+{apub} users are represented as a pair of `User` and `Identity` objects in each {zone}.
+These objects are created when a user first logs in on a {zone}.
+The user's Keycloak property `Username` is used to name the `User` object in each {zone}.
+
+{apub} organizations are represented as `Group` objects in each {zone}.
+The `Groups` are synchronized from Keycloak to all {zone}s in regular intervals.
+The group synchronization is implemented with the RedHat Communities of Practice https://github.com/redhat-cop/group-sync-operator[`group-sync-operator`].
+
+Organization memberships are represented in each {zone} by adding the user's `User` object name to the organization's `Group` object.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/architecture/openshift-idp-data-model.adoc:14]: I think it makes sense to create a diagram for this data model. Should be easily doable with Drawio or so.

File: docs/modules/ROOT/pages/references/architecture/keycloak-data-model.adoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+= Keycloak data model
+
+:apub: APPUiO Public 2.0
```
Comment (tobru) [inline docs/modules/ROOT/pages/references/architecture/keycloak-data-model.adoc:3]: What's the reason for that as it is duplicated in each page anyways? Wouldn't it make more sense to have this configure globally for all pages? Afaik OpenShift does it that way for the OpenShift documentation, but I don't know how.

**Review by tobru — APPROVED** [review]
LGTM, only a non-blocking comment and the missing diagram.

File: docs/modules/ROOT/pages/references/architecture/keycloak-data-model.adoc (hunk 2 above)
Comment (simu) [reply docs/modules/ROOT/pages/references/architecture/keycloak-data-model.adoc]: This attribute is now defined globally in `docs/antora.yml`.

**Review by corvus-ch — APPROVED** [review]

**Review by tobru — APPROVED** [review]

**Comment by ccremer** [comment]
I have pushed commits that adds the "global" data model, whereas the "identity data model" shows how just User and Organization entities are implemented.

**Pull request merged**